src/ol/interaction/Snap.js

```diff
@@ -510,6 +510,7 @@ class Snap extends PointerInteraction {
               closestVertex = vertex;
               minSquaredDistance = delta;
               closestFeature = segmentData.feature;
+              closestSegment = segmentData.segment;
```

Comment:
I don't think you want to return the closest segment when you've snapped to a vertex. Some users need a distinction between a vertex snap and a segment snap, and if you return the closest segment on a vertex snap it will be much harder to detect that. And it would be an undesired breaking change.

Reply:
Ok I have removed it.